### PR TITLE
adds fade-in animation

### PR DESCRIPTION
**Pull request by michelejoseph1**
### Summary:
_Context_ - This is part of a larger effort to create an Android app that takes in a list of ingredients and returns a list of cocktail recipes that can be made using them.

_Reference_ - [RapidAPI](https://rapidapi.com/thecocktaildb/api/the-cocktail-db) and [TheCocktailDB](https://www.thecocktaildb.com/api.php)

_This Diff_ - adds a fade-in animation so that items in recyclerview fade into the next activity (RecipeDetailsActivity)

### Test Plan:
https://user-images.githubusercontent.com/66920319/180861316-f96fdbeb-e92f-43ec-b0ce-ae9e5063e766.mov

### CODE REVIEW

File: app/src/main/res/anim/zoom.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        android:fromXScale="0.5"
+        android:toXScale="3.0"
+        android:fromYScale="0.5"
+        android:toYScale="3.0"
+        android:duration="4000"
+        android:pivotX="50%"
+        android:pivotY="50%" >
+    </scale>
+
+    <scale xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (alfonsocheng) [inline app/src/main/res/anim/zoom.xml:14]: not sure why you need the scale animation twice, was there a reason for it?

File: app/src/main/res/anim/zoom.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <scale xmlns:android="http://schemas.android.com/apk/res/android"
+        android:fromXScale="0.5"
+        android:toXScale="3.0"
+        android:fromYScale="0.5"
+        android:toYScale="3.0"
+        android:duration="4000"
```
Comment (alfonsocheng) [inline app/src/main/res/anim/zoom.xml:9]: despite the 4000 duration, the animation is looking really fast

File: app/src/main/res/anim/zoom.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        android:pivotX="50%"
+        android:pivotY="50%" >
+    </scale>
```
Comment (alfonsocheng) [inline app/src/main/res/anim/zoom.xml:12]: do you know if  adding will help smooth out the animation?
```
        android:interpolator="@android:anim/linear_interpolator"
```

File: app/src/main/java/com/example/cocktailme/CocktailNamesActivity.java [hunk 4]
```diff
@@ -65,6 +67,8 @@ protected void onCreate(Bundle savedInstanceState) {
         insertedIngredients = getIntent().getStringExtra("search");
         cocktails = new ArrayList<>();
         cocktailAdapter = new CocktailAdapter(this, cocktails);
+        Animation object = AnimationUtils.loadAnimation(getApplicationContext(), R.anim.zoom);
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/CocktailNamesActivity.java:70]: this looks un-used, can probably remove ?

File: app/src/main/java/com/example/cocktailme/adapters/CocktailAdapter.java [hunk 5]
```diff
@@ -74,9 +76,10 @@ public void onClick(View v) {
             int position = getAdapterPosition();
             if (position != RecyclerView.NO_POSITION) {
                 RecipeModel ingredient = recipeModels.get(position);
+                Bundle bndlanimation = ActivityOptions.makeCustomAnimation(context, R.anim.fade_in,R.anim.fade_out).toBundle();
                 Intent intent = new Intent(context, RecipeDetailsActivity.class);
                 intent.putExtra(RecipeModel.class.getName(), ingredient);
-                context.startActivity(intent);
+                context.startActivity(intent, bndlanimation);
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/adapters/CocktailAdapter.java:82]: nit: typo on bndle instead of bundle

File: app/src/main/java/com/example/cocktailme/RecipeDetailsActivity.java [hunk 6]
```diff
@@ -60,15 +60,23 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
+
     public void setRatingText(View v) {
         submitButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 Log.i(TAG, "entered onClick");
                 TextView t = (TextView) findViewById(R.id.avgRatingText);
-                t.setText("The average rating for this cocktail is: " + cocktail.getAverageRating() + "stars");
+               // t.setText("The average rating for this cocktail is: " + cocktail.getAverageRating() + "stars");
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/RecipeDetailsActivity.java:79]: any reason this was commented out?

File: app/src/main/java/com/example/cocktailme/RecipeDetailsActivity.java [hunk 7]
```diff
@@ -60,15 +60,23 @@ protected void onCreate(Bundle savedInstanceState) {
         cocktailID = recipeModel.getId();
         getInstructions(cocktailID);
 
+        cocktailImage.setOnLongClickListener(new View.OnLongClickListener() {
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/RecipeDetailsActivity.java:63]: oh nice! I would have probably put this in a separate PR

File: app/src/main/java/com/example/cocktailme/adapters/CacheAdapter.java [hunk 8]
```diff
@@ -79,4 +88,13 @@ public interface OnRecipeListener {
             void onRecipeClick(int position);
             void onHorizontalRecipeClick(int position);
         }
+        private void setAnimation(View viewToAnimate, int position)
```
Comment (alfonsocheng) [inline app/src/main/java/com/example/cocktailme/adapters/CacheAdapter.java:91]: is this still being used? looks like slide_in_left is removed already

File: app/src/main/java/com/example/cocktailme/RecipeDetailsActivity.java (hunk 6 above)
Comment (michelejoseph1) [reply app/src/main/java/com/example/cocktailme/RecipeDetailsActivity.java:72]: that line was giving me an error so i just commented it out

**Review by alfonsocheng — APPROVED** [review]
nice job!
